_posts/2020-04-27-introducing-command-mode.adoc

```diff
@@ -0,0 +1,111 @@
+
+== How to use it
+
+Below is a simple `GreetingMain` class which uses the traditional `GreetingService` from all our quickstarts.
+
+[source,java]
+----
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.annotations.QuarkusMain;
+
+@QuarkusMain    // <.>
+public class GreetingMain implements QuarkusApplication {
+  @Inject // <.>
+  GreetingService service;
+
+  @Override
+  public int run(String... args) throws Exception {   // <.>
+
+    if(arg.length>0) {
+      System.out.println(service(arg[0]);
+    } else P{
+      System.out.println(service(""));
+    }
+
+    return 0;
+ }
+}
+----
+<.> The `@QuarkusMain` annotation tells Quarkus that this is the main entry point.
+<.> The `run` method is invoked once Quarkus starts, and the application stops when it finishes. If you would like to access request scoped beans you could annotate it with `@ActivateRequestContext`.
+
+You can add this to a `GreetingMain.java` and compile to a .jar or to full native (using `mvnw package -Dnative`) and when run you get something like:
+
+```
+ ./target/getting-started-command-mode-1.0-SNAPSHOT-runner commando
+__  ____  __  _____   ___  __ ____  ______
+ --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) getting-started-command-mode 1.0-SNAPSHOT (powered by Quarkus 999-SNAPSHOT) started in 0.005s.
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) Profile prod activated.
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) Installed features: [cdi]
+hello commando
+2020-04-25 00:14:38,999 INFO  [io.quarkus] (main) getting-started-command-mode stopped in 0.000s
+```
+
+=== Development mode
+
+Additionally command mode also works with `quarkus:dev` aka. dev-mode.
+
+When you run with `mvn quarkus:dev` you can add `-Dquarkus.args=yourvalue` as arguments you want to pass into the command line. 
+
+Thus with `mvn quarkus:dev -Dquarkus.args=test` the app will start and after end look like this:
+
+```
+__  ____  __  _____   ___  __ ____  ______
+ --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+ -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+2020-04-25 00:26:13,916 INFO  [io.quarkus] (Quarkus Main Thread) getting-started-command-mode 1.0-SNAPSHOT (powered by Quarkus 1.4.0.Final) started in 0.644s.
+2020-04-25 00:26:13,934 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
+2020-04-25 00:26:13,934 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi]
+hello test
+2020-04-25 00:26:13,956 INFO  [io.quarkus] (Quarkus Main Thread) getting-started-command-mode stopped in 0.022s
+Quarkus application exited with code 0
+Press Enter to restart or Ctrl + C to quit
+```
+
+You can now Press Enter multiple times to force a rerun- and if you edited source code Press Enter will trigger build and then restart with no real overhead.
+
+=== Main methods
```

Comment:
So this is very unclear. Can you use the `main` method without a CLI / command mode. i.e. can I have my normal RESTEasy app be restarted from main. It think that's what people wanted.

Reply:
You can.